src/hooks/useTransactionSteps.ts

```diff
@@ -77,12 +93,12 @@ const useTransactionSteps = ({ transaction, onComplete }: Props) => {
     const sourceTransactionFinalized = stepEvaluator(transaction.block, bestBlockFinalized);
     const blockFinalityRelayed = stepEvaluator(transaction.block, bestBridgedFinalizedBlockOnTarget);
     const messageDelivered = stepEvaluator(transaction.messageNonce, latestReceivedNonceRuntimeApi);
-    const messageFinalizedOnTarget = stepEvaluator(transaction.messageNonce, targetMessageDeliveryBlock);
+    const messageFinalizedOnTarget = stepEvaluator(transactionNonceOfTargetFinalizedBlock, nonceOfCurrentTargetBlock);
```

Comment:
2) Comparing apples with pears. I was comparing the transaction message nonce with a block. 
I didn't catch this because when chains are restarted this conditions were satisfied.

Reply:
6 & END: the message is finalized once we verify that current nonce of the current block is bigger than the nonce obtained from the delivered message block.